smithy-syntax/src/main/java/software/amazon/smithy/syntax/TreeType.java

```diff
@@ -532,6 +532,7 @@ void parse(CapturingTokenizer tokenizer) {
         @Override
         void parse(CapturingTokenizer tokenizer) {
             tokenizer.withState(this, () -> {
+                tokenizer.expect(IdlToken.IDENTIFIER);
```

Comment:
Is this to capture "with"? In that case, we don't treat it as an IDENTIFIER since it's a literal string value.

Reply:
Yep, I confused myself with that one. Fixed it and removed the test case, since "with" is checked outside of the MIXINS production.